### DISCUSSION TITLE
String in a compound

### DISCUSSION THREAD

**Discussion by inter-otter** [BlueBrain/HighFive#984] [Questions] [answered]
I'm currently experimenting with compounds, starting with the code in `src/examples/compound_types.cpp`

I can build and use that, and I think I understand how that hangs together. I can extend it with additional data such as integers or fixed-size arrays, to read and write data.

Is it possible to somehow have a `std::string` member of a compound using this approach? When I try that I get a 'not trivially copyable' compilation error. I could make do with a fixed-size char array as a workaround, but I'd prefer to have a 'proper' string in the compound if that's feasible.

Thanks!

**Comment by 1uc** [comment, marked as answer]
It's one of the things we haven't implemented. For example there's this issue: https://github.com/BlueBrain/HighFive/issues/488

**Reply by inter-otter** [reply]
Ok, thanks for the information.
